requests/models.py

```diff
@@ -403,10 +404,16 @@ def prepare_url(self, url, params):
     def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
+        self.headers = CaseInsensitiveDict()
         if headers:
-            self.headers = CaseInsensitiveDict((to_native_string(name), value) for name, value in headers.items())
-        else:
-            self.headers = CaseInsensitiveDict()
+            #Regex checks that string doesn't begin with whitespace or contain return characters.
+            #Otherwise, it checks if it's a blank string.
+            pat = re.compile(r'^\S[^\r\n]*$|^$')
+            for name, val in headers.items():
+                name, val = to_native_string(name), to_native_string(val)
```

Comment:
I don't really mind if it gets moved. =)

As to the rb string syntax, in that case I recommend not using a raw string at all and just escaping the backslashes and using bytestrings.. It's annoying, but works on all platforms.